### ISSUE TITLE
infer types from super

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#289]

**Comment by DetachHead** [comment]
looks like pyright already does this, so all we need to do is make it not report `reportMissingParameterType` on the overridden method

**Comment by KotlinIsland** [comment]
even though pyright is working here, there is no inlay hint on the parameter

**Comment by DetachHead** [comment]
i think there still needs to be an option to still report `reportMissingParameterType` in such cases, because there are libraries that have decorators that rely on type annotations for runtime checks

```py
from pydantic import validate_call

class Foo:
    @abstractmethod
    def foo(self, value: int):
        ...

class Bar(Foo):
    @override
    @validate_call
    def foo(self, value):
        ...
```
ideally there would be something in the type system you could put on a decorator to tell the type checker that type annotations get runtime evaluated

**Comment by jorenham** [comment]
Yea that seems reasonable. It's in line with the current behavior:

https://pyright-play.net/?pythonVersion=3.13&strict=true&deprecateTypingAliases=true&code=GYJw9gtgBAhgRgYygSwgBzCALrOBnLEGBLCAUywAswATAKFEiiwE81kA7AcxXUxzAA3MiBDIaZOnQQAbGHjxQAYmDAAuOlC1QAAvAJES5KrU3aJwKMFUAKPGRnAANFEEwZAVzJqUHLAEoNbWCoADpwqVl5RQAhGBAbFTBAsy0dIRExCVSoCytbe0cXN08yQLCIkO0pOHioAF4oOIT-OlqQUOswGwAWACZW9s7bAGZQgEYe1rogA

https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBAhgRgYygSwgBzCALrOBnLEGBLCAUywAswATAKFEiiwE81kA7AcxXUxzAA3MiBDIaZOnQQAbGHjxQAYmDAAuOlC1QAAvAJES5KrU3aJwKMFUAKPGRnAANFEEwZAVzJqUHLAEoNbWCoADpwqVl5RQAhGBAbFTBAsy0dIRExCVSoCytbe0cXN08yQLCIkO0pOHioAF4oOIT-OlqQUOswGwAWACZW9s7bAGZQgEYe1rogA

**Comment by DetachHead** [comment]
yeah that's because mypy types it as `Any` if there's no annotation instead of inferring it from the base class like pyright does

```py
class Foo:
    @abstractmethod
    def foo(self, value: int):
        ...

class Bar(Foo):
    @override
    def foo(self, value):
        reveal_type(value) # Any
```

**Comment by jorenham** [comment]
> yeah that's because mypy types it as `Any` if there's no annotation instead of inferring it from the base class like pyright does

Ah I see.
I suppose that within the valid solution-space, mypy's would lie within the infemum. Or as Wolfgang Pauli [once put it](https://en.wikipedia.org/wiki/Not_even_wrong):

> That is not only not right; it is not even wrong.
